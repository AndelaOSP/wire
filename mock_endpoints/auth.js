const auth = (req, res, next) => {
  const token = req.query.token || req.headers['authorization'];
  if (token == 'null' || token == 'false') {
    res.status(401).send({
      error: true,
      message: 'Sorry. You might not be authorized.',
      status: 'error'
    });
  } else {
    return next();
  }
};

module.exports = auth;
