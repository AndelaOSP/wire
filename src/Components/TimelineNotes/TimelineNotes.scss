@import "../../theme";
@import url(https://fonts.googleapis.com/css?family=Source+Sans+Pro);
@import url(https://fonts.googleapis.com/css?family=Montserrat);

.notes-toggle {
  display: flex;
  width: 200px;
  color: #9ca4ab;
  justify-content: space-between;
  margin-right: 12vw;
  margin-left: auto;
  padding: 20px 0px;

  input {
    width: 70px;
    left: -5px;
  }

  div:nth-child(1) {
    width: 50px;
    left: -15px;
  }

  span.toggle-label.mine, span.toggle-label.all {
    color: #1273bc !important;
  }

  div:nth-child(2) {
    div:nth-child(1) {
      div:nth-child(2) {
        div:nth-child(1) {
          background: purple;
          width: auto !important;
        }
      }
    }
  }
}

.notes-container {
  height: 65vh !important;
  display: flex;
  justify-content: center;
  flex-direction: column;
  overflow-y: scroll;

  .notes-list {
    width: 70vw;
    margin: 8rem auto;
    max-width: 980px;
    .subheader {
      padding-left: 0 !important;
      font-weight: 900 !important;
    }

    .note-date {
      padding-left: 26px;
      font-weight: 1000 !important;
    }

    .notes-list-item {
      padding: 16px 26px 0px 26px !important;

      & > div {
        & > div {
          padding: 0 !important;
        }
      }

      .single-note-container {
        padding: 0;

        .note-header {
          margin: 0.5rem 0;
        }

        .timestamp {
          width: 202px;
          height: 22px;
          font-family: 'DIN Pro';
          font-size: 14px;
          font-weight: 900;
          font-style: normal;
          font-stretch: normal;
          line-height: 1.29;
          text-align: left;
          color: #000000;
        }

        .note-time-underline {
          width: 150px !important;
          height: 5px !important;
          background-color: #F1F6F8 !important;
          border-radius: 5px !important;
        }

        .note-container {
          height: 100%;
          width: 100%;
          border-radius: 5px;

          .note-content {
            p {
              margin-top: 20px;
              font-family: 'DIN Pro' !important;
              font-size: 14px;
              line-height: 2.17;
              font-weight: 300;
              color: #000000;
            }
          }
        }

        .note-actions {
          margin-top: 0.7rem;
          padding: 1rem 0;

          .note-action-edit,
          .note-action-archive {
            padding-right: 0.4rem;
            cursor: pointer;
            fill: rgba(0, 0, 0, 0.4) !important;
            height: 20px !important;
          }
        }
      }
    }

    .note-divider {
      margin-top: 45px;
      padding: 0 26px 0 26px !important;
      height: 1px;
      margin-left: 26px;
      margin-right: 20px;
      background: repeating-linear-gradient(to right, rgba(0, 0, 0, 0.3) 0, rgba(0, 0, 0, 0.3) 5px, transparent 5px, transparent 12px)
    }

    .archive-button {
      background-color: #1273bc !important;
    }

    .no-message {
      width: 244.2px;
      height: 106px;
      background-color: #f3f6f9;
      margin: 0 auto;
      border-radius: 5px;

      p {
        font-family: 'DIN Pro';
        font-size: 16px;
        line-height: 1.31;
        letter-spacing: 0.2px;
        color: #000000;
        text-align: center;
        padding: 2.5rem;
      }
    }
  }

  .message-container {
    position: fixed;
    bottom: 0px;
    display: flex;
    flex-direction: row;
    background-color: #fff;
    width: 75vw;
    display: flex;
    justify-content: space-between;
    align-items: center;
    padding-bottom: 50px;

    .attachment-icon {
      cursor: pointer;
      margin-left: calc(31vw - 380px);
      margin-right: 25px;
      margin-top: 15px;
      width: 24px;
      height: 24px;
      fill: rgba(0, 0, 0, 0.6) !important;
    }

    .message-input {
      flex: 1 1 auto;
      height: 60px;
      max-width: 980px;
      border-bottom: 1.6px solid black;

      .text-input {
        text-align: center;
        padding-top: 15px;
        width: 100% !important;

        input[type="text"] {
          font-size: 18px !important;
        }

        align-items: center;
      }
    }

    .message-icon {
      cursor: pointer;
      height: 24px;
      margin-top: 5px;
      margin-left: 5px;
      opacity: 0.7;
    }

    .at-icon {
      cursor: pointer;
      font-family: 'Source Sans Pro';
      font-size: 24px;
      margin-top: 15px;
      margin-left: 15px;
      opacity: 0.6;
    }

    .add-button {
      cursor: pointer;
      margin-top: 15px;
      margin-left: 25px;
      width: 130px;
      height: 40px;
      box-shadow: 1px 2px 3px 1px rgba(0, 0, 0, 0.1);
      border-radius: 8px;
      border-style: none;

      &:hover {
        transform: scale(1.1, 1.1);
        background-color: #f5f5f5;
      }
    }
  }
}

@media only screen and (max-width: 600px) {
  .message-container .message-input {
    // width: 50vw !important;
    margin-left: 0 !important;
  }
}
