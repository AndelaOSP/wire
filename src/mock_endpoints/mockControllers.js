const fuzzysearch = require('fuzzysearch');

let { incidents, chats, notes, users, statuses, levels, locations } = require('./mockData');

module.exports = {
  getIncident: incidentId => {
    let incident = incidents.find(incident => {
      return incident.id === incidentId;
    });
    incident['Assignee'] =
      users.find(user => {
        return user.id === incident.assigneeId;
      }) || null;
    incident['User'] =
      users.find(user => {
        return user.id === incident.userId;
      }) || null;
    incident['Status'] =
      statuses.find(status => {
        return status.id === incident.statusId;
      }) || null;
    incident['Level'] =
      levels.find(level => {
        return level.id === incident.levelId;
      }) || null;
    incident['Location'] = locations.find(location => {
      return location.id === incident.locationId;
    });
    return incident;
  },
  getIncidents: query => {
    return incidents.filter(incident => fuzzysearch(query, incident.subject.toLowerCase()) === true).map(incident => {
      incident['Assignee'] =
        users.find(user => {
          return user.id === incident.assigneeId;
        }) || null;
      incident['User'] =
        users.find(user => {
          return user.id === incident.userId;
        }) || null;
      incident['Status'] =
        statuses.find(status => {
          return status.id === incident.statusId;
        }) || null;
      incident['Level'] =
        levels.find(level => {
          return level.id === incident.levelId;
        }) || null;
      return incident;
    });
  },
  getIncidentNotes: incidentId => {
    let incidentNotes = notes.filter(note => {
      return note.incidentId === incidentId;
    });
    incidentNotes.map(
      note =>
        (note['User'] = users.find(user => {
          return user.id === note.userId;
        }))
    );
    return incidentNotes;
  },
  getIncidentChats: incidentId => {
    let incidentChats = chats.filter(chat => {
      return chat.incidentId === incidentId;
    });
    incidentChats.map(chat => {
      chat['User'] = users.find(user => {
        return user.id === chat.userId;
      });
    });
    return incidentChats;
  },
  changeStatus: (incidentId, statusId) => {
    let incident = incidents.find(incident => {
      return incident.id === parseInt(incidentId);
    });
    incident.statusId = statusId;
    incident['Assignee'] =
      users.find(user => {
        return user.id === incident.assigneeId;
      }) || null;
    incident['User'] =
      users.find(user => {
        return user.id === incident.userId;
      }) || null;
    incident['Status'] =
      statuses.find(status => {
        return status.id === statusId;
      }) || null;
    incident['Level'] =
      levels.find(level => {
        return level.id === incident.levelId;
      }) || null;
    incident['Location'] = locations.find(location => {
      return location.id === incident.locationId;
    });
    return incident;
  },
  changeAssignee: (incidentId, assigneeId) => {
    let incident = incidents.find(incident => {
      return incident.id === incidentId;
    });
    incident.assigneeId = assigneeId;
    incident['Assignee'] =
      users.find(user => {
        return user.id === assigneeId;
      }) || null;
    incident['User'] =
      users.find(user => {
        return user.id === incident.userId;
      }) || null;
    incident['Status'] =
      statuses.find(status => {
        return status.id === incident.statusId;
      }) || null;
    incident['Level'] =
      levels.find(level => {
        return level.id === incident.levelId;
      }) || null;
    incident['Location'] = locations.find(location => {
      return location.id === incident.locationId;
    });
    return incident;
  },
  addNote: (incidentId, userId, note) => {
    let newNote = {
      id: notes.length++,
      incidentId,
      userId,
      note
    };
    notes.push(newNote);
    newNote['User'] =
      users.find(user => {
        return user.id === newNote.userId;
      }) || {};
    return newNote;
  },
  editNote: (noteId, note) => {
    let noteToEdit = notes.find(note => {
      return note.id === noteId;
    });
    noteToEdit.note = note;
    noteToEdit['User'] = users.find(user => {
      return user.id === noteToEdit.userId;
    });
    return noteToEdit;
  },
  archiveNote: noteId => {
    return notes.filter(note => {
      return note.id !== noteId;
    });
  },
  getChat: chatId => {
    return chats.find(chat => {
      return chat.id === chatId;
    });
  },
  addChat: (incidentId, userId, chat) => {
    let newChat = {
      id: chats.length++,
      incidentId,
      userId,
      chat
    };
    chats.push(newChat);
    newChat['User'] =
      users.find(user => {
        return user.id === newChat.userId;
      }) || {};
    return newChat;
  },
  editChat: (chatId, chat) => {
    let chatToEdit = chats.find(chat => {
      return chat.id === chatId;
    });
    chatToEdit.chat = chat;
    chatToEdit['User'] = users.find(user => {
      return user.id === chatToEdit.userId;
    });
    return chatToEdit;
  },
  archiveChat: chatId => {
    return chats.filter(chat => {
      return chat.id !== chatId;
    });
  },
  getStaff: () => {
    return users.filter(user => {
      return user.roleId !== 1;
    });
  }
};
