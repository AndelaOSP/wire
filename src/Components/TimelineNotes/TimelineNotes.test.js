import React from 'react';
import { shallow } from 'enzyme';
import shallowToJSON from 'enzyme-to-json';

import TimelineNotes from './TimelineNotes.Component';
import { testIncidents } from '../../../mock_endpoints/mockData';

describe('Timeline Notes component', () => {
  let wrapper;
  let wrapperInstance;
  const props = {
    incident: testIncidents[0],
    addNote: jest.fn(),
    editNote: jest.fn(),
    archiveNote: jest.fn(),
  };

  beforeEach(() => {
    wrapper = shallow(<TimelineNotes {...props} />);
    wrapperInstance = wrapper.instance();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should change showArchiveDialog state when handleOpenArchiveDialog method is called', () => {
    wrapperInstance.handleOpenArchiveDialog({ note: 'none' }, 1);

    expect(wrapperInstance.state.showArchiveDialog).toBeTruthy();
    expect(wrapperInstance.state.note).toEqual({ note: 'none' });
    expect(wrapperInstance.state.index).toEqual(1);
  });

  it('should change showArchiveDialog state when handleCloseArchiveDialog method is called', () => {
    wrapper.setState({
      note: {
        note: 'None',
      },
      index: 1,
      showArchiveDialog: true,
    });
    wrapperInstance.handleCloseArchiveDialog();

    expect(wrapperInstance.state.showArchiveDialog).toBeFalsy();
    expect(wrapperInstance.state.note).toEqual({});
    expect(wrapperInstance.state.index).toBeNull();
  });

  it('should change showEditDialog state when handleOpenEditDialog method is called', () => {
    wrapperInstance.handleOpenEditDialog({ note: 'none' }, 1);

    expect(wrapperInstance.state.showEditDialog).toBeTruthy();
    expect(wrapperInstance.state.note).toEqual({ note: 'none' });
    expect(wrapperInstance.state.index).toEqual(1);
  });

  it('should change showEditDialog state when handleOpenEditDialog method is called', () => {
    wrapper.setState({
      showEditDialog: true,
      note: {
        note: 'None',
      },
      index: 1,
    });
    wrapperInstance.handleCloseEditDialog();

    expect(wrapperInstance.state.showEditDialog).toBeFalsy();
    expect(wrapperInstance.state.note).toEqual({});
    expect(wrapperInstance.state.index).toBeNull();
  });

  it('should change showArchiveDialog state when handleArchiveNote method is called', () => {
    wrapper.setState({
      note: {
        id: '2',
        note: 'Note',
      },
      index: 2,
      showArchiveDialog: true,
    });
    wrapperInstance.handleArchiveNote({
      preventDefault: jest.fn(),
    });
    
    expect(props.archiveNote).toHaveBeenLastCalledWith('2', 2);
    expect(wrapperInstance.state.showArchiveDialog).toBeFalsy();
  });

  it('should change content state when handleChange method is called', () => {
    wrapperInstance.handleChange({
      preventDefault: jest.fn(),
      target: {
        value: 'Note',
      },
    });
    
    expect(wrapperInstance.state.content).toEqual('Note');
  });
  
  it('should initiate addNote action when handleAddNote method is called', () => {
    wrapper.setState({
      content: 'Some notes',
    });
    wrapperInstance.handleAddNote({
      preventDefault: jest.fn(),
    });

    expect(props.addNote).toHaveBeenLastCalledWith('Some notes', props.incident.id);
    expect(wrapperInstance.state.content).toEqual('');
  });

  it('should change the note state when handleEditNoteChange method is called', () => {
    wrapper.setState({
      note: {
        id: '1',
        note: 'important notes',
      },
    });
    wrapperInstance.handleEditNoteChange({
      preventDefault: jest.fn(),
      target: {
        value: 'Detailed notes',
      },
    });

    expect(wrapperInstance.state.note.note).toEqual('Detailed notes');
  });

  it('should initiate editNote action when handleEditNote method is called', () => {
    wrapper.setState({
      note: {
        id: '1',
        note: 'default notes',
      },
      showEditDialog: true,
      index: 1,
    });
    wrapperInstance.handleEditNote({
      preventDefault: jest.fn(),
    });

    expect(props.editNote).toHaveBeenCalledWith('default notes', '1', 1);
    expect(wrapperInstance.state.showEditDialog).toBeFalsy();
  });

  it('should return MM Do YYYY h:mm date format', () => {
    const dateStringFormat = wrapperInstance.handleDateString('2018-02-13T15:58:06.202Z');

    expect(dateStringFormat).toEqual('Feb 13th 2018 at 6:58 pm');
  });

  it('should display a list of ListItem when an incident has notes', () => {
    wrapper.setProps({
      incident: {
        ...props.incident,
        notes: [
          {
            id: '1',
            note: 'some dummy notes',
          },
        ],
      },
    });

    expect(wrapper.find('ListItem').length).toEqual(1);
  });
});
