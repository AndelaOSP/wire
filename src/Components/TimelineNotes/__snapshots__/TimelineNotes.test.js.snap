// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timeline Notes component should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TimelineNotes
    addNote={[MockFunction]}
    archiveNote={[MockFunction]}
    editNote={[MockFunction]}
    incident={
      Object {
        "Level": Object {
          "name": "Red",
        },
        "Location": Object {
          "centre": "Nairobi",
          "country": "Kenya",
          "name": "Quiet Room",
        },
        "Status": Object {
          "status": "In Progress",
        },
        "assignees": Array [
          Object {
            "assignedRole": "ccd",
            "createdAt": "2018-03-20T15:45:48.778Z",
            "email": "caroline.nkirote@andela.com",
            "id": "-KhLsLqucbC1WAMylcFt",
            "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
            "roleId": 2,
            "updatedAt": "2018-03-20T15:45:48.778Z",
            "username": "Caroline Nkirote",
          },
          Object {
            "assignedRole": "assignee",
            "createdAt": "2018-03-20T15:45:48.778Z",
            "email": "mercy.muchai@andela.com",
            "id": "-KhLsOxrKcKZC8i2n888",
            "imageUrl": "https://lh3.googleusercontent.com/-XxYl2Ryrfns/AAAAAAAAAAI/AAAAAAAAABg/QJRPuQnerrk/photo.jpg",
            "roleId": 2,
            "updatedAt": "2018-03-20T15:45:48.778Z",
            "username": "Mercy Muchai",
          },
        ],
        "categoryId": 4,
        "chats": Array [],
        "createdAt": "2018-03-20T15:48:15.380Z",
        "dateOccurred": "2018-02-02T00:00:00.000Z",
        "description": "Some random thing happened and I didnt like it",
        "id": "cjezu2kr700010wx1yy6kxu5z",
        "levelId": 1,
        "locationId": "cjezu2kqo00000wx1a5u0i2xt",
        "notes": Array [],
        "reporter": Object {
          "createdAt": "2018-03-20T15:45:48.778Z",
          "email": "caroline.nkirote@andela.com",
          "id": "-KhLsLqucbC1WAMylcFt",
          "imageUrl": "https://lh4.googleusercontent.com/-rVipu2W1sBk/AAAAAAAAAAI/AAAAAAAAACQ/rOgW25IUgb8/photo.jpg",
          "roleId": 2,
          "updatedAt": "2018-03-20T15:45:48.778Z",
          "username": "Caroline Nkirote",
        },
        "statusId": 2,
        "subject": "Theft",
        "updatedAt": "2018-03-21T09:02:41.309Z",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <List
          className="notes-list"
        >
          <div
            className="no-message"
          >
            <p>
              No Notes Created
            </p>
          </div>
        </List>,
        <div
          className="message-container"
        >
          <img
            className="notification-icon"
            color="red"
            src="/assets/images/clip.svg"
          />
          <div
            className="message-input"
          >
            <form
              onSubmit={[Function]}
            >
              <TextField
                className="text-input"
                disabled={false}
                floatingLabelFixed={false}
                fullWidth={false}
                hintText="Add a note"
                multiLine={false}
                onChange={[Function]}
                rows={1}
                type="text"
                underlineShow={false}
                value=""
              />
            </form>
          </div>
          <div
            className="message-icon"
          >
            <img
              className="message-icon"
              src="/assets/images/smile.svg"
            />
          </div>
        </div>,
        <Dialog
          actions={
            Array [
              <CustomButton
                label="Cancel"
                onClick={[Function]}
              />,
              <CustomButton
                label="Archive"
                onClick={[Function]}
              />,
            ]
          }
          autoDetectWindowHeight={true}
          autoScrollBodyContent={false}
          modal={false}
          onRequestClose={[Function]}
          open={false}
          repositionOnUpdate={true}
        >
          Are you sure you want to archive this note?
        </Dialog>,
        <Dialog
          actions={
            Array [
              <CustomButton
                label="Cancel"
                onClick={[Function]}
              />,
              <CustomButton
                label="Submit"
                onClick={[Function]}
              />,
            ]
          }
          autoDetectWindowHeight={true}
          autoScrollBodyContent={false}
          modal={false}
          onRequestClose={[Function]}
          open={false}
          repositionOnUpdate={true}
          title="Edit note"
        >
          <TextField
            disabled={false}
            floatingLabelFixed={false}
            fullWidth={true}
            multiLine={true}
            onChange={[Function]}
            rows={3}
            type="text"
            underlineShow={true}
            value={undefined}
          />
        </Dialog>,
      ],
      "className": "notes-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <div
            className="no-message"
          >
            <p>
              No Notes Created
            </p>
          </div>,
          "className": "notes-list",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <p>
              No Notes Created
            </p>,
            "className": "no-message",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No Notes Created",
            },
            "ref": null,
            "rendered": "No Notes Created",
            "type": "p",
          },
          "type": "div",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              className="notification-icon"
              color="red"
              src="/assets/images/clip.svg"
            />,
            <div
              className="message-input"
            >
              <form
                onSubmit={[Function]}
              >
                <TextField
                  className="text-input"
                  disabled={false}
                  floatingLabelFixed={false}
                  fullWidth={false}
                  hintText="Add a note"
                  multiLine={false}
                  onChange={[Function]}
                  rows={1}
                  type="text"
                  underlineShow={false}
                  value=""
                />
              </form>
            </div>,
            <div
              className="message-icon"
            >
              <img
                className="message-icon"
                src="/assets/images/smile.svg"
              />
            </div>,
          ],
          "className": "message-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "notification-icon",
              "color": "red",
              "src": "/assets/images/clip.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <form
                onSubmit={[Function]}
              >
                <TextField
                  className="text-input"
                  disabled={false}
                  floatingLabelFixed={false}
                  fullWidth={false}
                  hintText="Add a note"
                  multiLine={false}
                  onChange={[Function]}
                  rows={1}
                  type="text"
                  underlineShow={false}
                  value=""
                />
              </form>,
              "className": "message-input",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <TextField
                  className="text-input"
                  disabled={false}
                  floatingLabelFixed={false}
                  fullWidth={false}
                  hintText="Add a note"
                  multiLine={false}
                  onChange={[Function]}
                  rows={1}
                  type="text"
                  underlineShow={false}
                  value=""
                />,
                "onSubmit": [Function],
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "className": "text-input",
                  "disabled": false,
                  "floatingLabelFixed": false,
                  "fullWidth": false,
                  "hintText": "Add a note",
                  "multiLine": false,
                  "onChange": [Function],
                  "rows": 1,
                  "type": "text",
                  "underlineShow": false,
                  "value": "",
                },
                "ref": "noteInput",
                "rendered": null,
                "type": [Function],
              },
              "type": "form",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                className="message-icon"
                src="/assets/images/smile.svg"
              />,
              "className": "message-icon",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "message-icon",
                "src": "/assets/images/smile.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "actions": Array [
            <CustomButton
              label="Cancel"
              onClick={[Function]}
            />,
            <CustomButton
              label="Archive"
              onClick={[Function]}
            />,
          ],
          "autoDetectWindowHeight": true,
          "autoScrollBodyContent": false,
          "children": "Are you sure you want to archive this note?",
          "modal": false,
          "onRequestClose": [Function],
          "open": false,
          "repositionOnUpdate": true,
        },
        "ref": null,
        "rendered": "Are you sure you want to archive this note?",
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "actions": Array [
            <CustomButton
              label="Cancel"
              onClick={[Function]}
            />,
            <CustomButton
              label="Submit"
              onClick={[Function]}
            />,
          ],
          "autoDetectWindowHeight": true,
          "autoScrollBodyContent": false,
          "children": <TextField
            disabled={false}
            floatingLabelFixed={false}
            fullWidth={true}
            multiLine={true}
            onChange={[Function]}
            rows={3}
            type="text"
            underlineShow={true}
            value={undefined}
          />,
          "modal": false,
          "onRequestClose": [Function],
          "open": false,
          "repositionOnUpdate": true,
          "title": "Edit note",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "floatingLabelFixed": false,
            "fullWidth": true,
            "multiLine": true,
            "onChange": [Function],
            "rows": 3,
            "type": "text",
            "underlineShow": true,
            "value": undefined,
          },
          "ref": "notesTextField",
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <List
            className="notes-list"
          >
            <div
              className="no-message"
            >
              <p>
                No Notes Created
              </p>
            </div>
          </List>,
          <div
            className="message-container"
          >
            <img
              className="notification-icon"
              color="red"
              src="/assets/images/clip.svg"
            />
            <div
              className="message-input"
            >
              <form
                onSubmit={[Function]}
              >
                <TextField
                  className="text-input"
                  disabled={false}
                  floatingLabelFixed={false}
                  fullWidth={false}
                  hintText="Add a note"
                  multiLine={false}
                  onChange={[Function]}
                  rows={1}
                  type="text"
                  underlineShow={false}
                  value=""
                />
              </form>
            </div>
            <div
              className="message-icon"
            >
              <img
                className="message-icon"
                src="/assets/images/smile.svg"
              />
            </div>
          </div>,
          <Dialog
            actions={
              Array [
                <CustomButton
                  label="Cancel"
                  onClick={[Function]}
                />,
                <CustomButton
                  label="Archive"
                  onClick={[Function]}
                />,
              ]
            }
            autoDetectWindowHeight={true}
            autoScrollBodyContent={false}
            modal={false}
            onRequestClose={[Function]}
            open={false}
            repositionOnUpdate={true}
          >
            Are you sure you want to archive this note?
          </Dialog>,
          <Dialog
            actions={
              Array [
                <CustomButton
                  label="Cancel"
                  onClick={[Function]}
                />,
                <CustomButton
                  label="Submit"
                  onClick={[Function]}
                />,
              ]
            }
            autoDetectWindowHeight={true}
            autoScrollBodyContent={false}
            modal={false}
            onRequestClose={[Function]}
            open={false}
            repositionOnUpdate={true}
            title="Edit note"
          >
            <TextField
              disabled={false}
              floatingLabelFixed={false}
              fullWidth={true}
              multiLine={true}
              onChange={[Function]}
              rows={3}
              type="text"
              underlineShow={true}
              value={undefined}
            />
          </Dialog>,
        ],
        "className": "notes-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <div
              className="no-message"
            >
              <p>
                No Notes Created
              </p>
            </div>,
            "className": "notes-list",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <p>
                No Notes Created
              </p>,
              "className": "no-message",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "No Notes Created",
              },
              "ref": null,
              "rendered": "No Notes Created",
              "type": "p",
            },
            "type": "div",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                className="notification-icon"
                color="red"
                src="/assets/images/clip.svg"
              />,
              <div
                className="message-input"
              >
                <form
                  onSubmit={[Function]}
                >
                  <TextField
                    className="text-input"
                    disabled={false}
                    floatingLabelFixed={false}
                    fullWidth={false}
                    hintText="Add a note"
                    multiLine={false}
                    onChange={[Function]}
                    rows={1}
                    type="text"
                    underlineShow={false}
                    value=""
                  />
                </form>
              </div>,
              <div
                className="message-icon"
              >
                <img
                  className="message-icon"
                  src="/assets/images/smile.svg"
                />
              </div>,
            ],
            "className": "message-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "notification-icon",
                "color": "red",
                "src": "/assets/images/clip.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <form
                  onSubmit={[Function]}
                >
                  <TextField
                    className="text-input"
                    disabled={false}
                    floatingLabelFixed={false}
                    fullWidth={false}
                    hintText="Add a note"
                    multiLine={false}
                    onChange={[Function]}
                    rows={1}
                    type="text"
                    underlineShow={false}
                    value=""
                  />
                </form>,
                "className": "message-input",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <TextField
                    className="text-input"
                    disabled={false}
                    floatingLabelFixed={false}
                    fullWidth={false}
                    hintText="Add a note"
                    multiLine={false}
                    onChange={[Function]}
                    rows={1}
                    type="text"
                    underlineShow={false}
                    value=""
                  />,
                  "onSubmit": [Function],
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "text-input",
                    "disabled": false,
                    "floatingLabelFixed": false,
                    "fullWidth": false,
                    "hintText": "Add a note",
                    "multiLine": false,
                    "onChange": [Function],
                    "rows": 1,
                    "type": "text",
                    "underlineShow": false,
                    "value": "",
                  },
                  "ref": "noteInput",
                  "rendered": null,
                  "type": [Function],
                },
                "type": "form",
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <img
                  className="message-icon"
                  src="/assets/images/smile.svg"
                />,
                "className": "message-icon",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "message-icon",
                  "src": "/assets/images/smile.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "actions": Array [
              <CustomButton
                label="Cancel"
                onClick={[Function]}
              />,
              <CustomButton
                label="Archive"
                onClick={[Function]}
              />,
            ],
            "autoDetectWindowHeight": true,
            "autoScrollBodyContent": false,
            "children": "Are you sure you want to archive this note?",
            "modal": false,
            "onRequestClose": [Function],
            "open": false,
            "repositionOnUpdate": true,
          },
          "ref": null,
          "rendered": "Are you sure you want to archive this note?",
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "actions": Array [
              <CustomButton
                label="Cancel"
                onClick={[Function]}
              />,
              <CustomButton
                label="Submit"
                onClick={[Function]}
              />,
            ],
            "autoDetectWindowHeight": true,
            "autoScrollBodyContent": false,
            "children": <TextField
              disabled={false}
              floatingLabelFixed={false}
              fullWidth={true}
              multiLine={true}
              onChange={[Function]}
              rows={3}
              type="text"
              underlineShow={true}
              value={undefined}
            />,
            "modal": false,
            "onRequestClose": [Function],
            "open": false,
            "repositionOnUpdate": true,
            "title": "Edit note",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "floatingLabelFixed": false,
              "fullWidth": true,
              "multiLine": true,
              "onChange": [Function],
              "rows": 3,
              "type": "text",
              "underlineShow": true,
              "value": undefined,
            },
            "ref": "notesTextField",
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
